File: ruby/fizzy.rb
#!/usr/bin/env ruby

require "rubygems"
require "parallel"
require "contracts"
include Contracts

Contract Num => String
def fizz(n)
  x, y = n % 3, n % 5

  if [x, y] == [0, 0]
    "FizzBuzz"
  elsif x == 0
    "Fizz"
  elsif y == 0
    "Buzz"
  else
    n.to_s
  end
end

def main
  Parallel.map(1.upto(100)) { |i| fizz(i) }.each { |s| puts s }
end

main if __FILE__ == $0

File: scripting/general/python/fizzy.py
#!/usr/bin/env python

'''FizzBuzz'''

# Backwards compatibility with Python 2;
# Python 3 allows print() in lambdas.
from __future__ import print_function

import multiprocessing
from contracts import *


@contract(n='int')
def fizzbuzz(n):
    '''FizzBuzz value of a given number'''

    x, y = n % 3, n % 5

    if (x, y) == (0, 0):
        return 'FizzBuzz'
    elif x == 0:
        return 'Fizz'
    elif y == 0:
        return 'Buzz'
    else:
        return str(n)


def main():
    '''Print FizzBuzz series'''

    p = multiprocessing.Pool()

    for s in p.map(
        fizzbuzz,
        range(1, 100)
    ):
        print(s)

if __name__ == '__main__':
    main()
